### PR TITLE
336636 secrets doc cleanup

### PR DESCRIPTION
**Pull request by akiaei01**
Cleaning up extra quotation marks and commas and fixing typos. 
Documenting slight behavior change - Any curly braces anywhere in the secret Id are not allowed (previously documented was leading or trailing curly braces)
Also added default value information to expirationDate and Description in Secrets.md as well as format examples for datetime

### CODE REVIEW

File: shared-content/management/secrets.md [hunk 1]
```diff
@@ -56,26 +56,25 @@ The following parameters are available for configuring secrets:
 
 | Parameter                 | Required | Type      | Description                                                  |
 | ------------------------- | -------- | --------- | ------------------------------------------------------------ |
-| **Id**              | Required | 'string' | Id of configuration to be added, edited, or removed. <br><br>Allowed value: any string that does not start or end with curly braces <br>(e.g., `<secretId>` is acceptable but `{{<secretId>}}`, `{<secretId>}`, or `{<secretId>` are not)|
-| **Description** | Optional | 'string' | Description of the secret. |
-| **ExpirationDate** | Optional | 'string' | Expiration date of the secret. |
+| **Id**              | Required | 'string' | Id of configuration to be added, edited, or removed. <br><br>Allowed value: any string that does not contain curly braces <br>(e.g., `<secretId>` is acceptable but `{{<secretId>}}`, `{<secretId>}`, or `{<secretId>` are not)|
```
Comment (aveva-jhwang) [inline shared-content/management/secrets.md:59]: I doubt anyone would have curly braces in the middle of the Id string but is that scenario supported? If so, maybe the wording for this should be: "any string that is not bound by curly braces"

File: shared-content/management/secrets.md [hunk 2]
```diff
@@ -56,26 +56,25 @@ The following parameters are available for configuring secrets:
...
+| **Description** | Optional | 'string' | Description of the secret.  <br><br>Allowed value: any string <br> Default: `null` <br> **Note:** The **Description** field is not used in the adapter and is metadata.|
+| **ExpirationDate** | Optional | 'datetime' | Expiration date of the secret. <br><br> Allowed value: any Json DateTime string <br> Examples of allowed formats:  UTC: `yyyy-mm-ddThh:mm:ssZ`; Local: `mm-dd-yyyy hh:mm:ss` <br>Default: `null`<br> **Note:** The **ExpirationDate** field is not used in the adapter and is metadata. |
```
Comment (aveva-jhwang) [inline shared-content/management/secrets.md:61]: I think the sentence flows better when you have it in this order: "The Description field is metadata and not used in the adapter." Same for the ExpirationDate field as well.

File: shared-content/management/secrets.md (hunk 1 above)
Comment (akiaei01) [reply shared-content/management/secrets.md:59]: This is what we mentioned in that long talk after standup, curly braces anywhere in the ID should not be allowed. so `Secret{Value`, `{SecretValue}`, or `SecretValue}` would all not be allowed.
Unless you see a use case for a user wanting to have a curly brace somewhere in the middle of the Id string?

File: shared-content/management/secrets.md (hunk 2 above)
Comment (aveva-jhwang) [inline shared-content/management/secrets.md:61]: Regarding the allowed formats for the date, are these the only formats supported? If so, I'd reword it to "Allowed formats: UTC ....". If others are allowed, I would link to the list of examples so that users can pick the one that works best for them.

File: shared-content/management/secrets.md (hunk 1 above)
Comment (aveva-jhwang) [reply shared-content/management/secrets.md:59]: I thought what you shared during standup was the case where a customer would have { } at the start and end of the Id string, not in between. Regardless, I still don't see any reason why customers would have curly braces in the middle of the string so I'm good with your changes. The examples listed could be adjusted to not make it seem like it's only for cases when you'd have it start and end with curly braces. Maybe that'd make the example too long...

File: shared-content/management/secrets.md [hunk 3]
```diff
@@ -56,26 +56,25 @@ The following parameters are available for configuring secrets:
...
 	{
 	  "Id": "OmfEgress.DataEndpoints.Endpoint1.ClientSecret",
-	  "Value": "CfDJ8FK3AvrP",
+	  "Value": "CfDJ8FK3AvrP"
 	},
 	{
 	  "Id": "MyCustomSecret",
 	  "Description": "This is a secret",
-	  "ExpirationDate": "06/17/2024 18:40:57"
+	  "ExpirationDate": "2024-06-17",
```
Comment (aveva-jhwang) [inline shared-content/management/secrets.md:77]: Is there a reason this example was changed? In the description in the table, the allowed format example includes time information and the date format for local time is different. If this is a supported datetime format, does it expire at 23:59:59?

File: shared-content/management/secrets.md (hunk 1 above)
Comment (akiaei01) [reply shared-content/management/secrets.md:59]: Hmm I think the wording is clear enough that there should not be any curly braces in the id string, and I'm not sure how to add another example that has a brace within it while still using the `<secretId>` format. What do you think @rmuolic ?

**Review by kshuOSIsoft — APPROVED** [review]

File: shared-content/management/secrets.md (hunk 3 above)
Comment (akiaei01) [reply shared-content/management/secrets.md:77]: Yes, the previous example does not actually work. If you do not specify the time, it defaults to 0, so `"2024-06-17T00:00:00"` in this case. 
Good catch, that was actually a copy-paste error from egress-endpoints.md, so I'll update that file as well. I just checked and putting a datetime in the format `mm-dd-yyyy hh:mm:ss` does not work in secrets or in egress-endpoints

File: shared-content/management/secrets.md (hunk 1 above)
Comment (akiaei01) [reply shared-content/management/secrets.md]: I made a slight change to add another not allowed example `<secret{Id>`, let me know what you think

**Review by aveva-jhwang — DISMISSED** [review]
I'm good with the changes. We can always adjust as we get feedback from our customers. Thanks!

File: shared-content/management/secrets.md [hunk 4]
```diff
@@ -56,26 +56,25 @@ The following parameters are available for configuring secrets:
...
-| **ExpirationDate** | Optional | 'string' | Expiration date of the secret. |
-| **Value** | Required | 'string' | The secret value. <br><br>Allowed value: any string that is not enclosed by double curly braces <br>(e.g.,  `<value>` is acceptable but `{{<value>}}` is not)|
+| **Id**              | Required | 'string' | Id of configuration to be added, edited, or removed. <br><br>Allowed value: any string that does not contain curly braces <br>(e.g., `<secretId>` is acceptable but `{{<secretId>}}`, `{<secretId>}`, `<secret{Id>`or `{<secretId>` are not)|
+| **Description** | Optional | 'string' | Description of the secret.  <br><br>Allowed value: any string <br> Default: `null` <br> **Note:** The **Description** field is metadata and not used in the adapter.|
+| **ExpirationDate** | Optional | 'datetime' | Expiration date of the secret. <br><br> Allowed formats:  UTC: `"yyyy-mm-ddThh:mm:ssZ"`; Local: `"yyyy-mm-ddThh:mm:ss"`. If the time not specified, it will default to the start of the day (e.g. `2025-06-19` will default to `2025-06-19T00:00:00`) <br>Default: `null`<br> **Note:** The **ExpirationDate** field is metadata and not used in the adapter. |
```
Comment (rmuolic) [inline shared-content/management/secrets.md:61]:
```suggestion
| **ExpirationDate** | Optional | 'datetime' | Expiration date of the secret. <br><br> Allowed formats:  UTC: `"yyyy-mm-ddThh:mm:ssZ"`; Local: `"yyyy-mm-ddThh:mm:ss"`. If the time is not specified, it will default to the start of the day (e.g. `2025-06-19` will default to `2025-06-19T00:00:00`) <br>Default: `null`<br> **Note:** The **ExpirationDate** field is metadata and not used in the adapter. |
```

File: shared-content/management/secrets.md [hunk 5]
```diff
@@ -56,26 +56,25 @@ The following parameters are available for configuring secrets:
...
+| **Value** | Required | 'string' | The secret value. <br><br>Allowed value: any string that is not explicitly enclosed by double curly braces <br>(e.g.,  `<value>`, `{{{<value>}}`, or `{{<val{ue>}}` is acceptable but `{{<value>}}` is not)|
```
Comment (rmuolic) [inline shared-content/management/secrets.md:62]:
```suggestion
| **Value** | Required | 'string' | The secret value. <br><br>Allowed value: any string that is not explicitly enclosed by double curly braces <br>(e.g., `<value>`, `{{{<value>}}`, or `{{<val{ue>}}` is acceptable but `{{<value>}}` is not)|
```

File: shared-content/management/secrets.md [hunk 6]
```diff
@@ -90,7 +89,7 @@ For the purposes of readability, the secret id is abbreviated {sid}.
 | api/v1/management/secrets | GET | Returns entire secrets configuration (secret values will be hidden by *s) |
 | api/v1/management/secrets | PUT | Creates or replaces entire secrets configuration |
 | api/v1/management/secrets | DELETE | Deletes entire secrets configuration |
-| api/v1/management/secrets/{sid} | GET | Retuns single secret if exists (secret values will be hidden by *s) |
-| api/v1/management/secrets/{sid} | PUT | Creates or repaces a single secret |
+| api/v1/management/secrets/{sid} | GET | Returns single secret if exists (secret values will be hidden by *s) |
```
Comment (rmuolic) [inline shared-content/management/secrets.md:92]:
```suggestion
| api/v1/management/secrets/{sid} | GET | Returns a single secret if it exists (secret values will be hidden by *s) |
```

File: shared-content/management/secrets/reference-secrets.md [hunk 7]
```diff
@@ -71,7 +69,9 @@ The secrets management configuration updates to add the following id and value:
 }
 ```
 
-The health configuration replaces `<clientsecret>` with the generated secret Id `"{{System.HealthEndpoints.OCS.ClientSecret}}"`:
+> **Note:** The **Value** `<clientsecret>` is encrypted before storing in the secrets management configuration.
```
Comment (rmuolic) [inline shared-content/management/secrets/reference-secrets.md:72]:
```suggestion
 **Note:** The **Value** `<clientsecret>` is encrypted before storing in the secrets management configuration.
```

File: shared-content/troubleshooting/troubleshooting.md [hunk 8]
```diff
@@ -104,7 +104,9 @@ Date and time strings should use the following formats:
 
 UTC: `yyyy-mm-ddThh:mm:ssZ`
 
-Local: `mm-dd-yyyy hh:mm:ss`
+Local: `yyyy-mm-ddThh:mm:ss`
+
+If the time not specified, it will default to the start of the day (e.g. `2025-06-19` will default to `2025-06-19T00:00:00`)
```
Comment (rmuolic) [inline shared-content/troubleshooting/troubleshooting.md:109]:
```suggestion
If the time not specified, it will default to the start of the day (e.g., `2025-06-19` will default to `2025-06-19T00:00:00`)
```

File: shared-content/configuration/health-endpoints.md [hunk 9]
```diff
@@ -61,7 +61,7 @@ The following parameters are available for configuring health endpoints:
...
-| **DebugExpiration** | Optional | string | Enables logging of detailed information, for each outbound HTTP request pertaining to this egress endpoint, to disk. The value represents the date and time this detailed information should stop being saved. Examples of valid strings representing date and time: UTC: "yyyy-mm-ddThh:mm:ssZ", Local: "mm-dd-yyyy hh:mm:ss". For more information, see {link to troubleshooting section}.<br><br>Default: `null` |
+| **DebugExpiration** | Optional | string | Enables logging of detailed information, for each outbound HTTP request pertaining to this egress endpoint, to disk. The value represents the date and time this detailed information should stop being saved. Examples of valid strings representing date and time: UTC: "yyyy-mm-ddThh:mm:ssZ", Local: "yyyy-mm-ddThh:mm:ss". For more information, see [Egress debug logging](xref:TroubleshootTheAdapter#egress-debug-logging)..<br><br>Default: `null` |
```
Comment (rmuolic) [inline shared-content/configuration/health-endpoints.md:64]:
```suggestion
| **DebugExpiration** | Optional | string | Enables logging of detailed information, for each outbound HTTP request pertaining to this egress endpoint, to disk. The value represents the date and this detailed information should stop being saved. Examples of valid strings representing date and time: UTC: "yyyy-mm-ddThh:mm:ssZ", Local: "yyyy-mm-ddThh:mm:ss". For more information, see [Egress debug logging](xref:TroubleshootTheAdapter#egress-debug-logging).<br><br>Default: `null` |
```

File: shared-content/configuration/egress-endpoints.md [hunk 10]
```diff
@@ -66,7 +66,7 @@ The following parameters are available for configuring egress endpoints:
...
+| **DebugExpiration**             | Optional                  | string    | Enables logging of detailed information to disk for each outbound HTTP request pertaining to the egress endpoint. The value represents the date and time this detailed information should stop being saved. The value represents the date and time this detailed information should stop being saved. Examples of valid strings representing date and time:  UTC: `yyyy-mm-ddThh:mm:ssZ`, Local: `yyyy-mm-ddThh:mm:ss`. For more information, see [Egress debug logging](xref:TroubleshootTheAdapter#egress-debug-logging).<br><br>Default: `null`|
```
Comment (rmuolic) [inline shared-content/configuration/egress-endpoints.md:69]:
```suggestion
| **DebugExpiration**             | Optional                  | string    | Enables logging of detailed information to disk for each outbound HTTP request pertaining to the egress endpoint. The value represents the date and time that this detailed information should stop being saved. Examples of valid strings representing date and time:  UTC: `yyyy-mm-ddThh:mm:ssZ`, Local: `yyyy-mm-ddThh:mm:ss`. For more information, see [Egress debug logging](xref:TroubleshootTheAdapter#egress-debug-logging).<br><br>Default: `null`|
```

**Review by rmuolic — CHANGES_REQUESTED** [review]
Minor corrections noted in suggestions.

**Review by rmuolic — APPROVED** [review]

**Review by kshuOSIsoft — APPROVED** [review]

**Review by aveva-jhwang — APPROVED** [review]
